modules/navier_stokes/src/postprocessors/InternalVolumetricFlowRate.C

```diff
@@ -0,0 +1,162 @@
+  params.addParam<MooseEnum>("advected_interp_method",
+                             advected_interp_method,
+                             "The interpolation to use for the advected quantity. Options are "
+                             "'upwind' and 'average', with the default being 'upwind'.");
+  params.addCoupledVar(
+      "advected_variable", 0, "The advected variable quantity of which to study the flow");
+  params.addParam<MaterialPropertyName>(
+      "advected_mat_prop", 0, "The advected material property of which to study the flow");
+  return params;
+}
+
+InternalVolumetricFlowRate::InternalVolumetricFlowRate(const InputParameters & parameters)
+  : InterfaceIntegralPostprocessor(parameters),
+    _fv(getParam<bool>("fv")),
+    _vel_x(coupledValue("vel_x")),
+    _vel_y(coupledValue("vel_y")),
+    _vel_z(coupledValue("vel_z")),
+    _fv_vel_x(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_x", 0))),
+    _fv_vel_y(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_y", 0))),
+    _fv_vel_z(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_z", 0))),
+    _advected_variable(coupledValue("advected_variable")),
+    _fv_advected_variable(
+        dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("advected_variable", 0))),
+    _advected_material_property(getADMaterialProperty<Real>("advected_mat_prop")),
+    _advected_material_property_neighbor(getNeighborADMaterialProperty<Real>("advected_mat_prop"))
+{
+  /// Check that at most one advected quantity has been provided
+  if (parameters.isParamSetByUser("advected_variable") &&
+      parameters.isParamSetByUser("advected_mat_prop"))
+    mooseError(
+        "InternalVolumetricFlowRatePostprocessor should be provided either an advected variable or "
+        "an advected material property");
+
+  using namespace Moose::FV;
+
+  const auto & advected_interp_method = getParam<MooseEnum>("advected_interp_method");
+  if (advected_interp_method == "average")
+    _advected_interp_method = InterpMethod::Average;
+  else if (advected_interp_method == "upwind")
+    _advected_interp_method = InterpMethod::Upwind;
+  else
+    mooseError("Unrecognized advected quantity interpolation type ",
+               static_cast<std::string>(advected_interp_method));
+}
+
+Real
+InternalVolumetricFlowRate::computeQpIntegral()
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  if (_fv)
+  {
+    /// We should not be at a boundary
+    const FaceInfo * const fi = _mesh.faceInfo(_current_elem, _current_side);
+    mooseAssert(fi, "We should have a face info");
+    mooseAssert(
+        !fi->isBoundary(),
+        "Use VolumetricFlowRate instead of InternalVolumetricFlowRate for domain boundaries");
+
+    const Elem * neighbor = _current_elem->neighbor_ptr(_current_side);
+
+    /// Get face value for velocity
+    // FIXME Make sure getInternalFaceValue uses the right interpolation method, see #16585
+    const auto & vx_face =
+        !getFieldVar("vel_x", 0)
```

Comment:
We can avoid calling this for every qp. In your constructor, you can initialize a boolean member like `_is_coupled_vel_x` with the result of `isCoupled("vel_x")`. Let's do this for all the velocity components

Reply:
```c++
InternalVolumetricFlowRate::InternalVolumetricFlowRate(const InputParameters & parameters)
  : InterfaceIntegralPostprocessor(parameters),
_is_coupled_vel_x(isCoupled("vel_x")),
_is_coupled_vel_y(isCoupled("vel_y")),
_is_coupled_vel_z(isCoupled("vel_z")),
```

Reply:
At some point, we will really want to implementing something such that we can avoid this ternary. Maybe it means that when we loop over internal faces in our postprocessor execution loop we populate a `MooseArray` like `_face_sln` in `MooseVariableFV` with the results of `getInternalFaceValue`. @rwcarlsen and I have talked about doing this for other quantities.

Reply:
I'm not going to make you do that here though. That's beyond the scope of this PR